src/main/java/org/snomed/snowstorm/fhir/services/FHIRValueSetService.java

```diff
@@ -1152,24 +1193,70 @@ public Parameters validateCode(String id, UriType url, UriType context, ValueSet
+				if(valueSetSystemMatchesRequestedSystem) {
+					String message = missing.getVersion() == null
+							? format("A definition for CodeSystem '%s' could not be found, so the code cannot be validated; Unable to check whether the code is in the value set '%s' because the code system %s was not found", requestedSystem, valueSetCanonical, requestedSystem)
+							: format("A definition for CodeSystem '%s' version '%s' could not be found, so the code cannot be validated. Valid versions: [%s]; Unable to check whether the code is in the value set '%s' because the code system %s was not found", requestedSystem, missing.getVersion(), availableVersion, valueSetCanonical, requestedSystem + "|" + missing.getVersion());
+					response.addParameter("message", message);
+					response.addParameter("result", false);
+					response.addParameter("system", new UriType(requestedSystem));
+					response.addParameter("version", missing.getVersion());
+					response.addParameter("x-caused-by-unknown-system", new CanonicalType(missing.toString()));
+					Extension e1 = new Extension("http://hl7.org/fhir/StructureDefinition/operationoutcome-message-id");
+					e1.setValue(new StringType("UNKNOWN_CODESYSTEM_VERSION"));
+					String ccText;
+					if(availableVersion!=null){
+						if(missing.getVersion() == null) {
+							ccText = format("A definition for CodeSystem '%s' could not be found, so the code cannot be validated. Valid versions: [%s]", requestedSystem, availableVersion);
+						} else {
+							ccText = format("A definition for CodeSystem '%s' version '%s' could not be found, so the code cannot be validated. Valid versions: [%s]", requestedSystem, missing.getVersion(), availableVersion);
+						}
+					} else {
+						ccText = format("A definition for CodeSystem '%s' could not be found, so the code cannot be validated", requestedSystem);
+					}
+					CodeableConcept detail1 = new CodeableConcept(new Coding(TX_ISSUE_TYPE, "not-found", null)).setText(ccText);
+					issues[0] = createOperationOutcomeIssueComponent(detail1,OperationOutcome.IssueSeverity.ERROR,null, e.getOperationOutcome().getIssueFirstRep().getCode(), List.of(e1),null);
+					String text = format("Unable to check whether the code is in the value set '%s' because the code system %s was not found", valueSetCanonical, missing.getVersion() == null ? requestedSystem : requestedSystem + "|" + missing.getVersion());
+					CodeableConcept detail2 = new CodeableConcept(new Coding(TX_ISSUE_TYPE, "vs-invalid", null)).setText(text);
+					Extension e2 = new Extension("http://hl7.org/fhir/StructureDefinition/operationoutcome-message-id");
+					e2.setValue(new StringType("UNABLE_TO_CHECK_IF_THE_PROVIDED_CODES_ARE_IN_THE_VALUE_SET_CS"));
+					issues[1] = createOperationOutcomeIssueComponent(detail2, OperationOutcome.IssueSeverity.WARNING, null, OperationOutcome.IssueType.NOTFOUND, List.of(e2), null);
+				} else {
```

Comment:
SonarQube is going to be asking me to break this method down to reduce complexity.   Leave it with me.